seatunnel-server/seatunnel-app/src/main/java/org/apache/seatunnel/app/service/impl/DatasourceServiceImpl.java

```diff
@@ -138,6 +145,13 @@ public String createDatasource(
         throw new SeatunnelException(SeatunnelErrorEnum.DATASOURCE_CREATE_FAILED);
     }
 
+    private void encryptDatasource(Map<String, String> datasourceConfig) {
+        datasourceConfig.replace(
+                Constants.PASSWORD,
+                ConfigShadeUtils.encryptOption(
+                        datasourceEncryptionType, datasourceConfig.get(Constants.PASSWORD)));
```

Comment:
password is not a mandatory field,  it can be empty also, then this code will throw NPE. Encrypt only if passord is not empty.